### ISSUE TITLE
Version conflict in spo version (16.0 vs. 16.1)

### ISSUE THREAD

**Issue by wulfland** [wulfland/SPOEmulators#2]
Nuget for 16 assemblkies must be compiled against the new nuget.

**Comment by mehrosu** [comment]
I am trying to look for package which can support CSOM 16

**Comment by wulfland** [comment]
In the [nuspec ](https://github.com/wulfland/SPOEmulators/blob/master/src/SPOEmulators/SPOEmulators.nuspec) is the dependency to 16.1
``<dependency id="Microsoft.SharePointOnline.CSOM" version="16.1.3912.1204"`/>`
You can fork the repo, uninstall the nuget, install the Version 16, Change this line. The package is created during build.

**Comment by mehrosu** [comment]
Hi 
I performed the below steps : 
1.I forked the repo. Tried to build solution it was giving error around  ShimAudit.
Audit Mask type.
1. To solve above said issue , I did excluded the code having ShimAudit & its further references.  Provided references of CSOM 16.0  reference from my machine already installed version. 
   Removed Pre Build events & Post Build events & generated related Fakes & Copied in another folder Fakes16.0.
   Reinstated Prebuild events  updated Fakes16 to Fakes 16.0 in the Pre Build Command event.
2. I tried updating in nuspec file.dependency id="Microsoft.SharePointOnline.CSOM" version="16.0.0.0" 
3. Updated Post Build event back finally the solution is building but with Post Build event script which is as far as I understood if for package creation  it is giving error.

Not sure am I missing something.

**Comment by wulfland** [comment]
The app.config of the test Project is expected, because it contains credentials to run tests against a o365 tenant. Copy the app.config.sample and rename it.
Which version of the nuget do yout want? I cant find a package for 16.0
![2016-02-04 13_56_12-spoemulators - microsoft visual studio administrator](https://cloud.githubusercontent.com/assets/5276337/12815562/57b72680-cb47-11e5-8bdf-a9d47b028550.png)

**Comment by mehrosu** [comment]
Yes we don't have that nugget package hence removed the dependency from nuspec.
The build is successful but  as soon as I try to build package with post build event its giving error.

Error   123 The command "if Release == Release (
C:\SPOEmulators-master\src..\lib\nuget.exe pack C:\SPOEmulators-master\src\SPOEmulators\SPOEmulators.nuspec -OutputDirectory C:\SPOEmulators-master\src\SPOEmulators\bin\Release\ -BasePath C:\SPOEmulators-master\src\SPOEmulators\ -verbose
)

if Release == Release15 (
C:\SPOEmulators-master\src..\lib\nuget.exe pack C:\SPOEmulators-master\src\SPOEmulators\SPOEmulators-OnPrem.nuspec  -OutputDirectory C:\SPOEmulators-master\src\SPOEmulators\bin\Release\ -BasePath C:\SPOEmulators-master\src\SPOEmulators\ -verbose
)" exited with code 1.

**Comment by mehrosu** [comment]
This is my new file 
<?xml version="1.0"?>
<package >
    <metadata>
        <id>$id$</id>
        <version>0.3.3-Pre</version>
        <title>$title$</title>
        <authors>$author$</authors>
        <owners>$author$</owners>
        <licenseUrl>https://github.com/wulfland/SPOEmulators/blob/master/LICENSE</licenseUrl>
        <projectUrl>https://github.com/wulfland/SPOEmulators</projectUrl>
        <iconUrl>https://github.com/wulfland/SPOEmulators/blob/master/src/SPOEmulators/SPOEmulators_Logo_32_32.png?raw=true</iconUrl>
        <requireLicenseAcceptance>false</requireLicenseAcceptance>
        <description>$description$</description>
        <releaseNotes>A initial draft with only a few emulated types.</releaseNotes>
        <copyright>Copyright 2015</copyright>
        <tags>SharePoint O365 Office365 Apps Fake CSOM</tags>

```
    <frameworkAssemblies>
        <frameworkAssembly assemblyName="System" targetFramework="" />
        <frameworkAssembly assemblyName="System.Core" targetFramework="" />
        <frameworkAssembly assemblyName="Microsoft.SharePoint.Client" targetFramework="" />
        <frameworkAssembly assemblyName="Microsoft.SharePoint.Client.Runtime" targetFramework="" />
    </frameworkAssemblies>
</metadata>
<files>
    <file src="tools\init.ps1" target="tools" />
    <file src="tools\install.ps1" target="tools" />
    <file src="Fakes17\Microsoft.SharePoint.Client.fakes" target="content\Fakes" />
    <file src="Fakes17\Microsoft.SharePoint.Client.Runtime.fakes" target="content\Fakes" />
    <file src="bin\Release\SPOEmulators.dll" target="lib\net45\SPOEmulators.dll" />
</files>
```

</package>

**Comment by wulfland** [comment]
I just created a new package for the newest version of the nuget and published it to nuget.org.
If you want to create your own package for the newest assemblies without the nuget you have to do the following:
- Update your project to .Net 4.6.1
- Update *.fake file:  
  ``<Assembly Name="Microsoft.SharePoint.Client"`Version="16.1.0.0"/>`
- Reference the new assemblies
  If the post build Fails you can create your package from the package Manager console:

`..\lib\nuget.exe pack .\SPOEmulators\SPOEmulators.csproj -OutputDirectory .\SPOEmulators\bin\Release -BasePath .\SPOEmulators -verbose`

But one question: why don't you use the nuget?

**Comment by mehrosu** [comment]
Yes it would have been much straight if I can use the package created by you.
My application is using  CSOM dll which are on version 16.0.0.0 which my team is not looking to upgrade.
We are using Visual Studio 2013 Ultimate Update 2.0 which I have just updated  to Update 5.  for Micrososft.Quality.Tools.Testing.Fakes version 12.0.

**Comment by wulfland** [comment]
Let me know when you Need more help...

**Comment by mehrosu** [comment]
Thank You.
Just now got my package build.
I had almost no experience in SharePoint development. I am working with team to support them to make the code testable. I hope I would be able to utilize this Emulator for my work.

**Comment by wulfland** [comment]
Perfect. Please provide Feedback. I used the Version for farm solutions (SPEmulators) a lot - but lately I haven't written much code myself. So I'm glad for every Feedback.

**Comment by mehrosu** [comment]
Hi I tried to use approach defined in Provisioning Engine for adding some List on web in fake mode.
As of now  no list is getting populated.

Just a reference I have called  Provisioning Engine CreateDefaultList 
Later I called connect.ClientContext.Web.Lists.GetByTitle for same title nothing returned.
 Further Investigation shows context.Clientontext.Lists in watcher shows  exception 
Provider    '((Microsoft.SharePoint.Client.ClientObjectCollection<Microsoft.SharePoint.Client.List>)(context.ClientContext.Web.Lists)).Provider' threw an exception of type 'System.NullReferenceException' System.Linq.IQueryProvider {System.NullReferenceException}

**Comment by wulfland** [comment]
I had this error one. When I remember it correctly it was connected with the visual Studio Version. Try right clicking mscorlib in you Project and click "Add Fake Assembly."
